pkg/controller/user/user_controller.go

```diff
@@ -242,15 +250,17 @@ func (c *userController) reconcile(key string) error {
 		// ensure user kubeconfig configmap is created
 		if err = c.kubeconfig.CreateKubeConfig(user); err != nil {
 			klog.Error(err)
+			c.recorder.Event(user, corev1.EventTypeWarning, controller.FailedSynced, fmt.Sprintf("Failed to sync: %s", err))
 			return err
 		}
 	}
 
 	if c.devopsClient != nil {
 		// assign jenkins role after user create, assign multiple times is allowed
 		// used as logged-in users can do anything
-		if err = c.assignDevOpsAdminRole(user); err != nil {
+		if err = c.waitForAssignDevOpsAdminRole(user); err != nil {
 			klog.Error(err)
+			c.recorder.Event(user, corev1.EventTypeWarning, controller.FailedSynced, fmt.Sprintf("Failed to sync: %s", err))
 			return err
```

Comment:
Should `return err` be removed? Otherwise, reconcile will run again.

Reply:
fixed